### PR TITLE
Make sure printing XLA tensor only execute the HLO once

### PR DESCRIPTION
**Pull request by JackCaoG**

### CODE REVIEW

File: test/test_operations.py [hunk 1]
```diff
@@ -1649,6 +1649,40 @@ def test_cached_addcdiv(self):
...
+
+    # case 1 mark_step
+    t1 = torch.randn(1, 4, device=xla_device)
+    xm.mark_step()
+    xm.wait_device_ops()
+    self.assertEqual(met.metric_data('ExecuteTime')[0], 1)
+    for _ in range(3):
+      print(t1)
+    self.assertEqual(met.metric_data('ExecuteTime')[0], 1)
+    self.assertIn('xla::device_data',
```
Comment (vanbasten23) [inline test/test_operations.py:1665]: I wonder what this`self.assertIn('xla::device_data'` is testing for.

File: test/test_operations.py [hunk 2]
```diff
@@ -1649,6 +1649,40 @@ def test_cached_addcdiv(self):
...
+  def test_print_executation(self):
+    xla_device = xm.xla_device()
+    xm.mark_step()
+    met.clear_all()
+
+    # case 1 mark_step
+    t1 = torch.randn(1, 4, device=xla_device)
+    xm.mark_step()
+    xm.wait_device_ops()
+    self.assertEqual(met.metric_data('ExecuteTime')[0], 1)
```
Comment (will-cromar) [inline test/test_operations.py:1661]: side note: we need a less obscure way to get metrics. I assume `met.metric_data(...)[0]` is the number of samples for a metric?

**Review by will-cromar — APPROVED** [review]
LGTM. Was there a recent bug that inspired this test?

**Review by vanbasten23 — APPROVED** [review]

File: test/test_operations.py (hunk 1 above)
Comment (JackCaoG) [reply test/test_operations.py:1665]: make sure t1 is a `device_data` after print, this mean execution has materialize the tensor.

**Review by alanwaketan — APPROVED** [review]
LGTM. Good news that we don't have bugs in the code.

**Comment by JackCaoG** [comment]
somehow cpu CI failed, it passed TPU CI on my end. let me try to repo.

File: test/test_operations.py (hunk 2 above)
Comment (wonjoolee95) [reply test/test_operations.py:1661]: +1 for this, I think it should be pretty simple, we probably just need to add some getter functions. Yeah currently 0th index is `TotalSamples`. Ex. output looks like:
```
Metric: OutboundData
  TotalSamples: 3
  Accumulator: 12.00B
  ValueRate: 34.00B / second
  Rate: 8.50056 / second
  Percentiles: 1%=4.00B; 5%=4.00B; 10%=4.00B; 20%=4.00B; 50%=4.00B; 80%=4.00B; 90%=4.00B; 95%=4.00B; 99%=4.00B
  ```

**Review by wonjoolee95 — APPROVED** [review]

**Comment by JackCaoG** [comment]
oh it failed for `XLA_USE_EAGER_DEBUG_MODE` which is expected. I would skip the test is we are in `XLA_USE_EAGER_DEBUG_MODE` since it will force each op to be execute right after it is called and messed up the `ExecuteTime` counter.

**Review by yeounoh — APPROVED** [review]
LGTM

File: test/test_operations.py (hunk 2 above)
Comment (JackCaoG) [reply test/test_operations.py:1669]: yea sg

Comment (wonjoolee95) [reply test/test_operations.py:1669]: Not blocking though, I actually had a local branch that somewhat does this l when I was debugging something else with metrics -- so I can probably follow-up with a PR.

**Pull request merged**